src/app/query-home/search-area/pins/from-pin/show-menu.ts

```diff
@@ -0,0 +1,34 @@
+import {showContextMenu} from "src/js/lib/System"
+import Current from "src/js/state/Current"
+import Pools from "src/js/state/Pools"
+import popupPosition from "../../popup-position"
+
+export const showMenu = (anchor: HTMLElement) => (dispatch, getState) => {
+  const s = getState()
+  const lakeId = Current.getLakeId(s)
+  const pools = Pools.getPools(lakeId)(s)
+
+  let selected = null
+  const template = pools
+    ? pools
+        .sort((a, b) => (a.name > b.name ? 1 : -1))
+        .map((p) => ({
+          label: p.name,
+          click: () => {
+            selected = p.name
```

Comment:
Mostly noting this as a reminder for myself: that by using the name here instead of the id, if someone changes the name of a pool then it will break their queries. On the other hand, by referring to the name instead of an internal/opaque id it makes the query easier to share where both parties have a pool with the same name.